dandi/dandiapi.py

```diff
@@ -325,24 +349,82 @@ class Version(APIBase):
     modified: datetime
 
 
-class RemoteDandiset(APIBase):
+class RemoteDandiset:
     """
     Representation of a Dandiset (as of a certain version) retrieved from the
     API
     """
 
-    client: "DandiAPIClient"
-    identifier: str
-    created: datetime
-    modified: datetime
-    #: The version in question of the Dandiset
-    version: Version
-    most_recent_published_version: Optional[Version]
-    draft_version: Version
+    def __init__(
+        self,
+        client: DandiAPIClient,
+        identifier: str,
+        version: Union[str, Version, None],
+        data: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        self.client = client
+        self.identifier = identifier
+        self._version_id: Optional[str]
+        self._version: Optional[Version]
+        if version is None:
+            self._version_id = None
+            self._version = None
+        elif isinstance(version, str):
+            self._version_id = version
+            self._version = None
+        else:
+            self._version_id = version.identifier
+            self._version = version
+        self._data = data
```

Comment:
given that those are "datasets", seeing `._data` suggests some notion of actual "data" (in our terms "assets") of the dataset, not its ... well, not "metadata" but may be , `._remote_rec` for "remote record" or just `._rec`?

Reply:
It's data about the Dandiset.  I think `_data` is perfectly fine.

Reply:
well, it is also in kwargs
```
 data: Optional[Dict[str, Any]] = None,
```
of a public interface and even without any documentation on what that `data` is.  So even if I agree that non-public `._data` is easy to understand the purpose of (which still isn't the case for me), having `data` in public interface is really confusing.

Also given that kwarg `data` is probably to be used only for internal purposes, why not to remove it from the interface and just make those internal "constructors" assign `._data` or have some internal `._set_data()` method?  Do you foresee use-case for a user to provide `data`?

Reply:
The constructors are not public interfaces.  None of them should be called by end users.

Reply:
Let's continue on that in https://github.com/dandi/dandi-cli/issues/731 since from current state it is not obvious on what is intended to be public and not